backends/ebpf/bpfinclude/ebpf_registry.c

```diff
@@ -0,0 +1,77 @@
+#include "ebpf_user.h"
+
+#define VAR_SIZE 32
+
+struct map_register {
+    char name[VAR_SIZE];        // name of the map
+    struct bpf_map_def *map;    // ptr to the map
+    UT_hash_handle hh;          // makes this structure hashable
+};
+
+struct map_register *maps = NULL;
+
+struct bpf_map_def *registry_lookup(const char *name) {
+    struct map_register *tmp_reg;
+    HASH_FIND(hh, maps, name, VAR_SIZE, tmp_reg);
+    if (tmp_reg==NULL)
+        return NULL;
+    return tmp_reg->map;
+}
+
+int registry_add(struct bpf_map_def *map) {
+    struct map_register *tmp_reg = (struct map_register*)malloc(sizeof(struct map_register));
+    HASH_ADD_KEYPTR(hh, maps, map->name, VAR_SIZE, tmp_reg);
```

Comment:
can this fail?
Do you always return 0?

Reply:
It can fail if the provided key in map is corrupt  (for example NULL or too long) or if malloc fails (unlikely). I can add checks for that.